my_library/models/library_book.py

```diff
@@ -1,15 +1,111 @@
+                   ('lost', 'Lost')
+                   ],
+        string='State'
+    )
+    description = fields.Html(string='Description')
+    cover = fields.Binary(string='Book Cover')
+    out_of_print = fields.Boolean(string='Out of Print?')
     date_release = fields.Date(string='Release Date')
-    author_ids = fields.Many2many(
+    date_update = fields.Datetime(string='Last Update', copy=False)
+    pages = fields.Integer(
+        string='Number of pages',
+        groups='base.group_user',
+        states={'lost': [('readonly', True)]},
+        help='Total book page count', company_dependent=False
+    )
+    reader_rating = fields.Float(string='Reader Average Rating', digits=(14, 4))
+    cost_price = fields.Float(string='Book Cost', digits=dp.get_precision('Book Price'))
+    currency_id = fields.Many2one(comodel_name='res.currency', string='Currency')
+    retail_price = fields.Monetary(string='Retail Price', currency_field='currency_id')
+    publisher_id = fields.Many2one(
         comodel_name='res.partner',
-        relation='book_partner',
-        column1='book_id',
-        column2='partner_id',
-        string='Authors'
+        string='Publisher',
+        ondelete='set null',
+        context={},
+        domain=[]
     )
+    publisher_city = fields.Char(string='Publisher City', related='publisher_id.city', readonly=True)
+    category_id = fields.Many2one(comodel_name='library.book.category')
+    _sql_constraints = [
+        ('name_uniq', 'UNIQUE (name)', 'Book title must be unique.'),
+        ('positive_page', 'CHECK(pages>0)', 'No of pages must be positive')
+    ]
+    age_days = fields.Float(
+        string='Days Since Release',
+        compute='_compute_days',
+        inverse='_inverse_age',
+        search='_search_age',
+        store=False,  # optional
+        compute_sudo=True  # optional
+    )
+
+    ref_doc_id = fields.Reference(selection='_referencable_models', string='Reference Document')
+
+    def name_get(self):
+        result = []
+        for record in self:
+            rec_name = "%s %s" % (record.name, "(%s)" % record.date_release if record.date_release else '')
+            result.append((record.id, rec_name))
+        return result
+
+    @api.constrains('date_release')
+    def _check_release_date(self):
+        for record in self:
+            if record.date_release > fields.Date.today():
+                raise ValidationError('Release date must be in the past')
+
+    @api.depends('date_release')
+    def _compute_age(self):
+        today = fields.Date.today()
+        for book in self:
+            if book.date_release:
+                delta = today - book.date_release
+                book.age_days = delta.days
+            else:
+                book.age_days = 0
+
+    def _inverse_age(self):
+        today = fields.Date.today()
+        for book in self.filtered('date_release'):
+            d = today - timedelta(days=book.age_days)
+            book.date_release = d
```

Comment:
book.date_release = today - timedelta(days=book.age_days)

or better

book.write({
    'date_release': today - timedelta(days=book.age_days),
   ...
   # you can add more, it will be written in one transaction!
})